### PR TITLE
Trap property tests

### PR DESCRIPTION
**Pull request by lastmjs**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
     "compilerOptions": {
         "strict": true,
         "target": "ES2020",
+        "module": "ES2020",
```
Comment (bdemann) [inline tsconfig.json:5]: out of curiosity why do we need this?

File: test/index.ts [hunk 2]
```diff
@@ -115,3 +117,23 @@ function processEnvVars(): {
         shouldRunBenchmarks
     };
 }
+
+export const defaultParams = {
+    numRuns: Number(process.env.AZLE_PROPTEST_NUM_RUNS ?? 1),
+    endOnFailure: process.env.AZLE_PROPTEST_SHRINK === 'true' ? false : true
+};
+
+export async function getCanisterActor<T>(
```
Comment (bdemann) [inline test/index.ts:126]: should we provide a way to add an agent?

File: test/index.ts [hunk 3]
```diff
@@ -115,3 +117,23 @@ function processEnvVars(): {
         shouldRunBenchmarks
     };
 }
+
+export const defaultParams = {
```
Comment (bdemann) [inline test/index.ts:121]: should this be defaultPropertyTestParams or something to make it clear they are for prop tests?

File: tests/property/ic_api/trap/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
+// TODO to improve these tests we would also test init, postUpgrade, and preUpgrade
```
Comment (bdemann) [inline tests/property/ic_api/trap/src/index.ts:1]: could you briefly explain why you didn't

**Review by bdemann — CHANGES_REQUESTED** [review]
Should I be worried about how many tests are failing?

File: tests/property/ic_api/trap/src/index.ts (hunk 4 above)
Comment (lastmjs) [reply tests/property/ic_api/trap/src/index.ts:1]: I'll just remove this commented code

File: test/index.ts (hunk 3 above)
Comment (lastmjs) [reply test/index.ts:121]: Fixed

File: tests/property/ic_api/trap/src/index.ts (hunk 4 above)
Comment (lastmjs) [reply tests/property/ic_api/trap/src/index.ts:1]: Removed

**Review by bdemann — DISMISSED** [review]

**Comment by lastmjs** [comment]
Trap has a problem, it isn't terminating. I am looking into it

**Comment by lastmjs** [comment]
I just needed to update a name I had changed in a previous PR

**Pull request merged**